src/lib/stitching/exchange/stitching.ts

```diff
@@ -0,0 +1,222 @@
+import { GraphQLSchema } from "graphql"
+import { amountSDL, amount } from "schema/fields/money"
+import gql from "lib/gql"
+
+const orderTotals = [
+  "itemsTotal",
+  "sellerTotal",
+  "commissionFee",
+  "totalListPrice",
+  "buyerTotal",
+  "taxTotal",
+  "shippingTotal",
+  "transactionFee",
+]
+const orderTotalsSDL = orderTotals.map(amountSDL)
+
+const lineItemTotals = ["shippingTotal", "listPrice", "commissionFee"]
+const lineItemTotalsSDL = lineItemTotals.map(amountSDL)
+
+const offerAmountFields = ["amount", "taxTotal", "shippingTotal", "buyerTotal"]
+const offerAmountFieldsSDL = offerAmountFields.map(amountSDL)
+export const exchangeStitchingEnvironment = (
+  localSchema: GraphQLSchema,
+  exchangeSchema: GraphQLSchema
+) => {
+  type DetailsFactoryInput = { from: string; to: string }
+
+  /**
+   * This returns a resolver which can take an exchange OrderParty union
+   * and convert it into a metaphysics `OrderParty`.
+   *
+   * You pass in the field to get the details from, and and the new fieldName
+   */
+  const partyUnionToDetailsFactory = ({ from, to }: DetailsFactoryInput) => {
+    // We abuse the query alias feature to make sure that all
+    // the data we need to generate the full object from
+    // is included.
+
+    // It's possible that this working around a bug.
+    const aliasedPartyFragment = (field, alias) => {
+      return gql`
+      ... on CommerceOrder {
+        ${alias}: ${field} {
+          __typename
+          ... on CommerceUser {
+            __typename
+            id
+          }
+          ... on CommercePartner {
+            __typename
+            id
+          }
+        }
+      }`
+    }
```

Comment:
This closure isn't using any of the in-scope variables, so might as well move its definition outside of the `partyUnionToDetailsFactory` function.

Reply:
I get ya, but I don't think it adds much value doing that given that it just moves the function that holds it to be 2 expressions